include/kframework/lemmas/bytes-simplification.k

```diff
@@ -7,33 +7,19 @@ module BYTES-SIMPLIFICATION [symbolic]
   // Buffer Reasoning
   // ########################
 
-    rule 0 <=Int lengthBytes ( _ ) => true [smt-lemma, simplification]
+    rule BA:Bytes       ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                    [simplification, concrete(BA)]
+    rule #buf(32, DATA) ==K BA:Bytes       =>          DATA  ==Int #asInteger(BA) requires lengthBytes(BA) <=Int 32 [simplification, concrete(BA)]
 
-    rule BA:Bytes   ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                    [simplification, concrete(BA)]
-    rule #buf(32, DATA) ==K BA:Bytes   =>          DATA  ==Int #asInteger(BA) requires lengthBytes(BA) <=Int 32 [simplification, concrete(BA)]
+    rule [bytes-concat-empty-right]: B:Bytes +Bytes .Bytes  => B [simplification]
+    rule [bytes-concat-empty-left]:  .Bytes  +Bytes B:Bytes => B [simplification]
 
-    rule #range(M, N1 +Int N2, K) => #range(#range(M, N2, lengthBytes(M) -Int N2), N1, K)
-      requires 0 <=Int N1 andBool 0 <=Int N2
-      [simplification, concrete(N2), concrete(M)]
-
-    rule #range(A +Bytes _B, START, LENGTH) => #range(A, START, LENGTH)
-      requires START +Int LENGTH <=Int lengthBytes(A) [simplification]
+    rule [bytes-concat-assoc-right]: (BA1 +Bytes BA2) +Bytes BA3 => BA1 +Bytes (BA2 +Bytes BA3) [simplification]
 
-    rule #range(A +Bytes B, START, LENGTH) => #range(B, START -Int lengthBytes(A), LENGTH)
-      requires lengthBytes(A) <=Int START [simplification]
-
-    rule #range(BA:Bytes, 0, LEN) => BA requires lengthBytes(BA) ==Int LEN [simplification]
-
-    rule (#range(#buf(32, X), 0, 28) ):Bytes => #buf(28, 0) requires #rangeUInt ( 32 , X ) [simplification]
-
-    rule (#range(#range( BA:Bytes, S0, L0), S1, L1) ) =>#range(BA, S0 +Int S1, minInt(L1, L0 -Int S1)) +Bytes #buf(maxInt(0, L1 -Int (L0 -Int S1)), 0)
-      requires 0 <=Int S0 andBool 0 <=Int L0 andBool 0 <=Int S1 andBool 0 <=Int L1 [simplification]
+    // #buf
 
-    rule WS         +Bytes .Bytes => WS  [simplification]
-    rule .Bytes +Bytes WS         => WS  [simplification]
-
-    // Re-associating bytes concatenation into a canonical right-associative form
-    rule (BA1 +Bytes BA2) +Bytes BA3 => BA1 +Bytes (BA2 +Bytes BA3) [simplification]
+    // TODO: custom ==K unification doesn't work in Haskell yet
+    // +Bytes unification
+    rule #buf(N, A) +Bytes BUF1 ==K #buf(N, B) +Bytes BUF2 => #buf(N, A) ==K #buf(N, B) andBool BUF1 ==K BUF2     [simplification]
```

Comment:
We do have a lemma that does this better, but I guess it will come in the subsequent PRs.

Reply:
I'll leave changes to the lemmas for subsequent PRs.